map/apply-scheduled-updates-to-preplanned-map-area/src/main/java/com/esri/samples/apply_scheduled_updates_to_preplanned_map_area/ApplyScheduledUpdatesToPreplannedMapAreaSample.java

```diff
@@ -0,0 +1,274 @@
+                      // set the parameters to download all updates for the mobile map packages
+                      offlineMapSyncParameters.setPreplannedScheduledUpdatesOption(PreplannedScheduledUpdatesOption.DOWNLOAD_ALL_UPDATES);
+                      // set the map package to rollback to the old state should the sync job fail
+                      offlineMapSyncParameters.setRollbackOnFailure(true);
+
+                      // create a sync job using the parameters
+                      OfflineMapSyncJob offlineMapSyncJob = offlineMapSyncTask.syncOfflineMap(offlineMapSyncParameters);
+
+                      // start the job and get the results
+                      offlineMapSyncJob.start();
+                      offlineMapSyncJob.addJobDoneListener(() -> {
+                        if (offlineMapSyncJob.getStatus() == Job.Status.SUCCEEDED) {
+                          OfflineMapSyncResult offlineMapSyncResult = offlineMapSyncJob.getResult();
+
+                          // if mobile map package reopen is required, close the existing mobile map package and load it again
+                          if (offlineMapSyncResult.isMobileMapPackageReopenRequired()) {
+                            mobileMapPackage.close();
+                            mobileMapPackage.loadAsync();
+                            mobileMapPackage.addDoneLoadingListener(() -> {
+                              if (mobileMapPackage.getLoadStatus() == LoadStatus.LOADED && !mobileMapPackage.getMaps().isEmpty()) {
+
+                                // add the map from the mobile map package to the map view
+                                mapView.setMap(mobileMapPackage.getMaps().get(0));
+
+                              } else {
+                                new Alert(Alert.AlertType.ERROR, "Failed to load the mobile map package.").show();
+                              }
+                            });
+                          }
+
+                          // perform an additional check for updates
+                          ListenableFuture<OfflineMapUpdatesInfo> offlineMapUpdatesInfoFuture2 = offlineMapSyncTask.checkForUpdatesAsync();
+                          offlineMapUpdatesInfoFuture2.addDoneListener(() -> {
+                            try {
+                              // get and check the results
+                              OfflineMapUpdatesInfo offlineMapUpdatesInfo2 = offlineMapUpdatesInfoFuture2.get();
+
+                              // update UI for available updates
+                              if (offlineMapUpdatesInfo2.getDownloadAvailability() == OfflineUpdateAvailability.AVAILABLE) {
+                                updateAvailableLabel.setText("Updates: AVAILABLE");
+
+                                // check and show update size
+                                long updateSize2 = offlineMapUpdatesInfo2.getScheduledUpdatesDownloadSize();
```

Comment:
should avoid using duplicate variable names with numbers after: though in this instance it's probably a handy indicator this code is being reused elsewhere and can be abstracted out into a method!

Reply:
Refactored as discussed!